src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -264,6 +271,49 @@ private async Task<SearchResult> RunSearch(SqlSearchOptions sqlSearchOptions, Ca
                     return await SearchImpl(sqlSearchOptions, _reuseQueryPlans.IsEnabled(_sqlRetryService), cancellationToken);
                 }
             }
+            else if (_queryConfiguration.DynamicSqlQueryPlanSelectionEnabled)
+            {
+                string hash = string.Empty;
+                bool reuseQueryCachingPlan = false;
```

Comment:
The default value should be set with `_reuseQueryPlans.IsEnabled(_sqlRetryService)`
false isn't the default for all services.

Reply:
Done. I created the concept of QueryPlanCacheLoader. It's a singleton class encapsulating the initialization of Query Plan Cache.